File: code/csharp/DesignPatterns/command/CommandClient.cs
namespace Command;
using System;

public class CommandClient
{
    public static void CommandExample()
    {
        // Create receiver (kitchen staff)
        Receiver chef = new Receiver();

        // Create concrete commands (orders)
        ICommand preparePizzaCommand = new PreparePizzaCommand(chef);
        ICommand servePastaCommand = new ServePastaCommand(chef);

        // Create invoker (waiter)
        Waiter waiter = new Waiter();

        // Set and execute commands
        waiter.Command = preparePizzaCommand;
        waiter.ExecuteCommand();

        waiter.Command = servePastaCommand;
        waiter.ExecuteCommand();
    }
}

File: code/typescript/src/command/CommandClient.ts
import { PreparePizzaCommand } from "./PreparePizzaCommand";
import { ServePastaCommand } from "./ServePastaCommand";
import { Receiver } from "./Receiver";
import { Waiter } from "./Waiter";

const chef = new Receiver();
const preparePizzaCommand = new PreparePizzaCommand(chef);
const servePastaCommand = new ServePastaCommand(chef);
const waiter = new Waiter();

waiter.setCommand(preparePizzaCommand);
waiter.executeCommand();

waiter.setCommand(servePastaCommand);
waiter.executeCommand();
